jmclient/jmclient/wallet_service.py

```diff
@@ -498,14 +510,26 @@ def sync_unspent(self):
         our_unspent_list = [x for x in unspent_list if (
             self.bci.is_address_labeled(x, wallet_name) or
             self.bci.is_address_labeled(x, self.EXTERNAL_WALLET_LABEL))]
-        for u in our_unspent_list:
-            if not self.is_known_addr(u['address']):
+        for utxo in our_unspent_list:
+            if not self.is_known_addr(utxo['address']):
                 continue
-            self._add_unspent_utxo(u, current_blockheight)
+            if "height" in utxo:
```

Comment:
So do I understand this correctly:
1/ the utxo dict returned from listunspent does actually (usually? almost always?) contain a "height" field, so that can and should be used directly where it's available. That part certainly makes sense.
2/ You've moved all the logic in _add_unspent_utxo for calculating and setting the height, back out into the calling function? But why do that, surely it'd be more logical to keep that logic in that function, since we might need to call it elsewhere?

Reply:
The two questions go together. The result of Bitcoin Core's `listunspent `RPC does not have a `height` field, instead it has `confirmations`. The result of `scantxoutset` does not have `confirmations` but only `height`. BitcoinCoreNoHistoryInterface actually isn't calling Core's `listunspent`, but returning its own dict that was the result of `scantxoutset`. So calculating and setting the height isn't needed if BitcoinCoreNoHistoryInterface is used, which is why that code is moved out of `_add_unspent_txo()`.
I considered obtaining the current blockchain height inside `BitcoinCoreNoHistoryInterface.rpc("listunspent")`, but decided against as it seemed silly to obtain the blockchain height to calculate confirmations, only later for the code to obtain blockchain height again to calculate the UTXO height. But I'm happy to reconsider if you think the code is confusing this way. Though for sure I should add a comment in the code explaining this.

Reply:
Got it. That makes a lot of sense. So now `WalletService.add_unspent_txo()` is just a thin wrapper (for format conversion) around `wallet.add_utxo()` which is fine, and the parsing logic of rpc call --> utxo information is all in the `sync_unspent()`.